### ISSUE TITLE
Proxy URL encoding 500 error on wiki pages edit with white spaces

### ISSUE THREAD

**Issue by Conspir3D** [go-gitea/gitea#3467]
- Gitea version (or commit ref):
- Git version: 2.1.4
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X] Not relevant
- Log gist:

## Description
I ran gitea behind a Apache Prox. When i edit wiki pages of a repository which contains Wheitespaces they get onverted to "-". So far, so good.

Behind the proxy ifyou edit a wiki page containing a whitespace in it's name, you get a 500 error on save and the log says:

`2018/02/06 10:59:49 [...routers/repo/wiki.go:500 EditWikiPost()] [E] EditWikiPage: Failed to remove /opt/gitea/data/tmp/local-wiki/4/Add%2BEmail%2BAddress.md: remove /opt/gitea/data/tmp/local-wiki/4/Add%2BEmail%2BAddress.md: no such file or directory`

this exists:
`/opt/gitea/data/tmp/local-wiki/4/Add-Email-Address.md`

I already added "nocanon" to the proxy configuration but that doesn't help:

```
<VirtualHost *:443>
        ProxyPreserveHost On
        ProxyRequests on
        ProxyPass / http://localhost:3000/ nocanon
        ProxyPassReverse / http://localhost:3000/

SSLEngine on
SSLCertificateFile /etc/letsencrypt/live/xxx.com/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/xxx.com/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf

        ServerName dev.xxx.com
        ServerAlias dev
</VirtualHost>
```

It's not reproducible on try.gitea.io as it doesn't run behind a proxy ;)

**Comment by lafriks** [comment]
On try.gitea.io it actually runs behind proxy, only not apache but caddy

**Comment by Conspir3D** [comment]
Oh, thx for that info but i couldn't even login nor register on try.gitea.io as it get's also 500 errors on registration and login, even to try to login with github account ;(

**Comment by lafriks** [comment]
try again, demo server was out of disk space, cleaned up a bit ;)

**Comment by michaelkuhn** [comment]
I can confirm this problem. Just noticed this in our instance (1.3.2+3-g88c363f9 behind an Apache proxy).

**Comment by Conspir3D** [comment]
thx for the confirmation, therefore i won't try to reproduce this on try.

Can now be labled as a bug ;)

**Comment by Conspir3D** [comment]
in release 
1.4.0+rc1-6-g221e502 the editing issue is fixed, BUT the link generation if e.g. [[link to any page]] is now buggy as white spaces are translated still to '+' instead of '-' which breaks the whole navigation at the wiki

**Comment by Conspir3D** [comment]
thx for adding it to 1.4 .. as i just saw, it's the last issue before 1.4 release ;)

**Comment by lafriks** [comment]
Closing as original issue for editing is fixed already in 1.4. For link generation there is separate issue #3506 already with PR to fix it.
